apollo-router/src/query_planner/fetch.rs

```diff
@@ -364,13 +385,44 @@ impl FetchNode {
             parameters.supergraph_request,
             parameters.schema,
             &self.input_rewrites,
+            &self.context_rewrites,
         ) {
             Some(variables) => variables,
             None => {
                 return (Value::Object(Object::default()), Vec::new());
             }
         };
 
+        let alias_query_string; // this exists outside the if block to allow the as_str() to be longer lived
+        let aliased_operation = if let Some(ctx_arg) = contextual_arguments {
+            // TODO: subgraph_schemas should be a HashMap of NodeStr or Name.
```

Comment:
I think it would be a good followup in general for the router to not work with `String`s but `NodeStr` or `Name` for any structures generated by federation_next or the router-bridge.

It's definitely not in the scope of this PR though!

Reply:
That was a @o0Ignition0o comment. Was there something more to do here?